--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino-1Ge4GW.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino-1Ge4GW.dts
@@ -95,3 +95,27 @@
 		firmware-postfix = "olinuxino";
 	};
 };
+
+&spi0 {
+	status = "okay";
+
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		status = "okay";
+	};
+};
+
+&uart3_pins {
+	pins = "PH4", "PH5";
+};
+
+&uart3 {
+	status = "okay";
+	pinctrl-0 = <&uart3_pins>;
+};
+
+&i2c1 {
+	status = "okay";
+};
